### DISCUSSION TITLE
24.12.14 백엔드 회의

### DISCUSSION THREAD

**Discussion by zeus6768** [woowacourse-teams/2024-code-zap#965] [이야기해봐요]
## 1. DTO 변수명 리팩토링
⚠️ 만장일치로 하기로 결정 ~!
예) CreateTemplateRequest → request

## 2. 리팩토링 전 통지 방식
⚠️ PR 메시지에 명시하자.
예) 켬미 코드 제가 바꿨으니 확인해주세요!

### 각자의 주장 
몰리: 싹다 var를 쓰자. 그게 일관성이 있다. 
제우스: 생성자와 정팩메는 타입이 명시적으로 보이기 때문에 var를 사용할 때의 단점이 없다. 
켬미: 프로덕션 코드에서 var를 사용해서 코드가 간결해진다고 해도 그게 큰 장점이라고 생각하지 않는다. 그러나 테스트에서는 var의 타입 추론 기능이 유용하다고 생각한다. 
짱수: 타입 추론이 가능한 대상에 대해서는 모두 var를 사용하자. repository.save() 메서드도 포함.

긴 논의 끝에 쪼금식 해보자고 결론 내렸습니다. 

⚠️ **생성자와 정팩메에 대해서는 var를 꼭 사용하기로 결정~~**

### EC2 2대 → RDS 1대로 변경
이렇게 하면 비용을 절감할 수 있다. 

근데 몰리 경험상 프리티어로 사용했는데도 비용이 나왔다고 해서... 추가 자료조사를 해야한다. 

⚠️ 몰리가 다음주 목요일까지 하는걸로 결정!

### ELB 사용
각 WAS 인스턴스에서 따로 SSL 인증서를 관리하지 않아도 됨 ❗

## 5. modifiedAt 수정 PR

⚠️ BaseTimeEntity를 상속한 SkipModifiedAtUpdateBaseTimeEnity 클래스 만들기로 결정

## 6. 카테고리 순서 (ordinal 필드) 누가 작업할지

⚠️ 19일 다음 주 목요일까지 PR 초롱이 하기로 결정!

## 7. 관리자 같이할사람!

제우스, 켬미

⚠️ 다음 주 목요일까지 이슈 작성

## 8. 인수테스트(통합테스트) 작성

⚠️ 이슈만 파고 아직 할당은 안 하기로 결정

**Comment by jminkkk** [comment]
인수 테스트 (통합 테스트) 작성하자~!

**Comment by kyum-q** [comment]
- disable 테스트 제가 확인하고 수정해봐도될까요?
   - [이슈](https://github.com/woowacourse-teams/2024-code-zap/issues/948#issuecomment-2530755602)와 같이 적어두려고 하는데 괜찮을까요? 
   - 그리고 수정하면서 사용하지 않는 서비스 코드나 레퍼지토리 코드 제거해도 될까요? (지우게 된다면 이슈에 추가할게요)
